src/main/java/pl/put/poznan/checker/logic/KeywordStepCountingVisitor.java

```diff
@@ -1,18 +1,60 @@
 package pl.put.poznan.checker.logic;
 
 public class KeywordStepCountingVisitor implements ScenarioVisitor {
+
+    public String[] keywords = {"ELSE", "IF", "FOR EACH"};
+
+    private boolean checkIfbeginsWithKeyword (String text) {
+        String[] splitStep = text.split("[ :]");
+
+        for (String keyword : this.keywords) {
+            if (splitStep[0].equals(keyword)){
+
+                return true;
+            }
+
+            if ((splitStep[0] + " " + splitStep[1]).equals(keyword)){
+
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+
     @Override
     public int visit(SimpleStep simpleStep) {
-        return 0;
+        if (checkIfbeginsWithKeyword(simpleStep.text)){
+            return 1;
+        }
+        else{
+            return 0;
+        }
     }
 
     @Override
     public int visit(ComplexStep complexStep) {
-        return 0;
+        int stepCount = 0;
+
+        for (Step st : complexStep.subscenario) {
+            stepCount += st.accept(this);
+        }
+
+        if (checkIfbeginsWithKeyword(complexStep.text)){
+            stepCount += 1;
+        }
+
+        return stepCount;
     }
 
     @Override
     public int visit(MainScenario mainScenario) {
-        return 0;
+        int stepCount = 0;
+
+        for (Step step : mainScenario.steps) {
+            stepCount += step.accept(this);
+        }
+        return stepCount;
     }
```

Comment:
I think we need to rethink the way we implemented the visitor.
According to specs, visit and accept function should be void instead of int.
For reference see my Pull Request #5  where I changed the way visitor is implemented to meet the void function requirement.

This wouldn't be big of a problem for current functionality but the next two functionalities from product backlog are working with strings, not numbers. Thats why I think we should consider changing it.

I'm open for a discussion, tho.

Reply:
Agreed, the visitor implementation can't stay as it is. I've been considering ot make it's method generic but making it void seems more elegant. I'd go for it.

Reply:
I'll work on this branch as soon as the #5 will be accepted and merged into main.